crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/builder/ContributionDTOBuilder.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.laa.crime.contribution.builder;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.justice.laa.crime.contribution.dto.ContributionDTO;
+import uk.gov.justice.laa.crime.contribution.model.CreateContributionRequest;
+
+@Component
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ContributionDTOBuilder {
+
+    public static ContributionDTO build(final CreateContributionRequest request) {
+        ContributionDTO.ContributionDTOBuilder builder = ContributionDTO.builder()
+                .applId(request.getApplId())
+                .repId(request.getRepId())
+                .contributionFileId(request.getContributionFileId())
+                .effectiveDate(request.getEffectiveDate() != null ? request.getEffectiveDate().toLocalDate() : null)
+                .calcDate(request.getCalcDate() != null ? request.getCalcDate().toLocalDate() : null)
```

Comment:
We need to map magCourtOutcome and repOrderDTO. Both the attributes are required by compare contribution